cpp/dolfinx/fem/assemble_matrix_impl.h

```diff
@@ -279,13 +278,13 @@ void assemble_interior_facets(
 
   // Temporaries for joint dofmaps
   std::vector<std::int32_t> dmapjoint0, dmapjoint1;
-
-  for (std::size_t index = 0; index < facets.size(); ++index)
+  assert(facets.size() % 4 == 0);
+  std::array<std::int32_t, 2> cells;
+  std::array<std::int32_t, 2> local_facet;
+  for (std::size_t index = 0; index < facets.size() / 4; ++index)
```

Comment:
`for (std::size_t index = 0; index < facets.size(); index += 4)` and simplifications below.